src/opnmidi_midiplay.cpp

```diff
@@ -1211,9 +1212,18 @@ void OPNMIDIplay::noteUpdate(size_t midCh,
                 if(vibrato && (!d || d->vibdelay_us >= m_midiChannels[midCh].vibdelay_us))
                     bend += static_cast<double>(vibrato) * m_midiChannels[midCh].vibdepth * std::sin(m_midiChannels[midCh].vibpos);
 
-#define BEND_COEFFICIENT 321.88557
-                synth.noteOn(c, BEND_COEFFICIENT * std::exp(0.057762265 * (currentTone + bend + phase)));
-#undef BEND_COEFFICIENT
+                double coef;
+                switch (synth.chipFamily()) {
+                    default:
+                    case OPNChip_OPN2:
+                        coef = 321.88557;
+                        break;
+                    case OPNChip_OPNA:
+                        coef = 309.12412;
+                        break;
```

Comment:
Maybe put those coefficients into side of OPN Family template than keep here?

Reply:
I don't know. I've forgotten how this value is computed, so I scaled it down by a ratio of chip's sampling rates and it worked.
I haven't understood for sure if this constant is more chip-related than it's implementation-related to ADLMIDI, about its value domain for pseudo-frequency values.

Reply:
In a case of here, it's just a constant, so, you can keep it internally, but comment a note about it for just in a case, my mistake I didn't put it into separated place, but just in the code here...

Reply:
Done. It's the same thing in libADLMIDI and original ADLMIDI, it's a hardcoded value at this place.